libcontainer/cgroups/fs/apply_raw.go

```diff
@@ -179,11 +180,24 @@ func (m *Manager) GetStats() (*cgroups.Stats, error) {
 }
 
 func (m *Manager) Set(container *configs.Config) error {
-	for name, path := range m.Paths {
-		sys, err := subsystems.Get(name)
-		if err == errSubsystemDoesNotExist || !cgroups.PathExists(path) {
+	for _, sys := range subsystems {
+		// We can't set this here, because after being applied, memcg doesn't
+		// allow a non-empty cgroup from having its limits changed.
+		if sys.Name() == "memory" {
```

Comment:
Essentially the workflow is as follows:
1. Create cgroups
2. Set the appropriate limits
3. Apply cgroups to the init process.

As of now `1` and `3` are essentially together. We should clean that up soon.